src/test/java/wolox/training/controllersTest/BookControllerTest.java

```diff
@@ -0,0 +1,168 @@
+    void whenUpdateBookThenReturnStatusCreated() throws Exception {
+        given(bookRepository.findById(1L)).withFailMessage(ErrorConstants.SUCCESS_CORRECT);
+        String json = new ObjectMapper().writeValueAsString(bookTest);
+        String url = (PATH + "/1");
+        mvc.perform(put(url)
+                .contentType(MediaType.APPLICATION_JSON)
+                .characterEncoding("utf-8")
+                .content(json))
+                .andDo(print())
+                .andExpect(status().isOk());
+    }
+
+    @WithMockUser(value = "pablo")
+    @Test
+    @DisplayName("Test, book is updated , it return status No Found")
+    void whenUpdateBookThenReturnStatusNoFound() throws Exception {
+        given(bookRepository.findById(1L)).withFailMessage(ErrorConstants.NOT_EXIST_ID);
+        String json = new ObjectMapper().writeValueAsString(bookTest);
+        String url = (PATH + "/1");
+        mvc.perform(put(url)
+                .contentType(MediaType.APPLICATION_JSON)
+                .characterEncoding("utf-8")
+                .content(json))
+                .andDo(print())
+                .andExpect(status().isNotFound());
+    }
+
+    @WithMockUser(value = "pablo")
+    @Test
+    @DisplayName("Test, book is deleted , it return status No Content")
+    void whenDeleteBookThenReturnStatusNoContent() throws Exception {
+        given(bookRepository.findById(1L)).withFailMessage(ErrorConstants.NOT_EXIST_ID);
+        String url = (PATH + "/1");
+        mvc.perform(delete(url)
+                .contentType(MediaType.APPLICATION_JSON))
+                .andDo(print())
+                .andExpect(status().isNoContent());
+    }
+
+    @WithMockUser(value = "pablo")
+    @Test
+    @DisplayName("Test, When a book is deleted , it return status No Found")
+    void whenDeleteBookThenReturnStatusNoFound() throws Exception {
+        given(bookRepository.findById(1L)).withFailMessage(ErrorConstants.NOT_EXIST_ID);
+        String url = (PATH + "/1");
+        mvc.perform(delete(url)
+                .contentType(MediaType.APPLICATION_JSON))
+                .andDo(print())
+                .andExpect(status().isNotFound());
+    }
+
+    @WithMockUser(value = "pablo")
+    @Test
+    @DisplayName("Test , find info by different param ,it return status OK")
+    void whenFindByAllParametersThenReturnStatusOK() throws Exception {
+        Pageable pageable = PageRequest.of(1, 4);
+        List<Book> books = new ArrayList<>();
+        books.add(bookTest);
+        Page<Book> bookPage = new PageImpl<>(books);
+        given(bookRepository.findByAllParameters("genre", "author", "image", "title", "subtitle", "publisher", "startYear", "endYear", 450, "2020", pageable)).withFailMessage(ErrorConstants.NOT_EXIST_ID);
+        String url = (PATH + "?genre=genre&author=author&image=image&title=title&subtitle=subtitle&publisher=publisher&startYear=10&endYear=2019&pages=22&isbn=22&page=1&size=4");
```

Comment:
This should be a constant. This should clean the line

Reply:
I changed it to constant value